src/extension/utils/createAndOpenConfFile.ts

```diff
@@ -1,30 +1,28 @@
 import { workspace, Uri, window } from 'vscode'
 import { log, Sources } from '../utils/log'
-import { InputFormData } from '../types'
-
-type ConfFile = {
-  files?: string[]
-  verify?: [string]
-  solc?: string
-  link?: string[]
-  settings?: string[]
-  staging?: string
-  cache?: string
-  msg?: string
-} & Record<string, boolean | string>
+import { InputFormData, ConfFile } from '../types'
 
 function setAdditionalSetting(val?: string) {
   if (val === 'true' || !val) return true
   if (val === 'false') return false
-
+  if (/^[0-9]+$/.exec(val)) return Number(val)
+  const mapRegex = /^{(".+":".+")(,".+":".+")*}/g
+  if (mapRegex.exec(val)) {
+    return JSON.parse(val)
+  }
+  // for --settings flags:
+  const settingsRegex = /(-(.+)=(.+))(,(-(.+)=(.+)))*/g
```

Comment:
There may be boolean flags too. Here's an example: `-boolean_flag,-another_flag=with_value`